app/controllers/tool_waitlists_controller.rb

```diff
@@ -0,0 +1,60 @@
+class ToolWaitlistsController < ApplicationController
+  # Can't use load_and_authorize_resource until it's fixed in the
+  # checkout controller as well.
+  # load_and_authorize_resource
+
+  # GET /tool_waitlists/new
+  def new
+    @tool_waitlist = ToolWaitlist.new
+    @tool_type = ToolType.find_by_id(params[:tool_type_id])
+    @tool_waitlist.tool_type = @tool
+  end
+
+  # POST /tool_waitlists
+  # POST /tool_waitlists.json
+  def create
+    @tool_waitlist = ToolWaitlist.new({
+        participant_id: params[:participant_id],
+        organization_id: params[:organization_id],
```

Comment:
Why are you keeping id's rather than doing Participant.find(params[:participant_id]) ?